src/service/resolver/amazon/amazon-reroute.service.ts

```diff
@@ -0,0 +1,119 @@
+import { URL } from 'url';
+
+import { ResolverException } from './../../../model/error/resolver-exception';
+import { AmazonReroute } from './../../../model/resolver/amazon-reroute';
+import { StatisticsService } from './../../statistics/statistic.service';
+import { AmazonCaptchaResolverService } from './amazon-captcha-resolver.service';
+
+export class AmazonRerouteService {
+  private static readonly AMAZON_HOSTS: string[] = [
+    'www.amazon.com',
+    'www.amazon.co.uk',
+    'www.amazon.ca',
+    'www.amazon.com.au',
+    'www.amazon.it',
+    'www.amazon.de',
+    'www.amazon.es',
+    'www.amazon.fr',
+  ];
+
+  private statisticsService: StatisticsService;
+
+  private reroutedRequests: Map<string, string[]>;
+
+  constructor(statisticsService: StatisticsService) {
+    this.statisticsService = statisticsService;
+    this.reroutedRequests = new Map<string, string[]>();
+  }
+
+  /**
+   * Check if should reroute because of captcha
+   * @param url Original URL
+   * @param error Error obtained
+   * @returns Should reroute
+   */
+  checkCaptcha(url: URL, error: ResolverException): AmazonReroute {
+    let result: AmazonReroute = AmazonReroute.noReroute();
+
+    if (
+      error != undefined &&
+      error.message == AmazonCaptchaResolverService.CAPTCHA_ERROR
+    ) {
+      console.error(error.message);
```

Comment:
Move this part to a function and the stuff should be done only when there is an URL.
Maybe the `changeHost` could change to `getReroute` and only if `shouldReroute()` is true do the console.error and stats changes